src/pmdas/bpftrace/bpftrace/process_manager.py

```diff
@@ -81,14 +81,16 @@ async def stop_bpftrace_process(self, script: Script, script_tasks: ScriptTasks)
         process.send_signal(signal.SIGINT)
 
         # wait max. 5s for graceful termination of the bpftrace process
-        _done, pending = await asyncio.wait({process.wait()}, timeout=5)
-        if pending:
+        try:
+            await asyncio.wait_for(process.wait(), timeout=5)
+        except TimeoutExpired:
```

Comment:
Afaics this exception is only thrown with the `subprocess` module (https://docs.python.org/3/library/subprocess.html#subprocess.TimeoutExpired) but not with the async version of it (`asyncio.subprocess.Process`, https://docs.python.org/3/library/asyncio-subprocess.html#asyncio.subprocess.Process.wait).

Reply:
Yep, thanks - after further reading I've switched over to what I believe is the correct exception now.